include/cuco/detail/static_set/kernels.cuh

```diff
@@ -89,6 +90,257 @@ CUCO_KERNEL void find(InputIt first, cuco::detail::index_type n, OutputIt output
   }
 }
 
+template <typename CG,
+          typename Size,  // have to do this due to https://github.com/NVIDIA/cccl/issues/1523
+          typename ProbeKey,
+          typename Key,
+          typename AtomicT,
+          typename OutputIt1,
+          typename OutputIt2>
+__device__ void flush_buffer(CG const& tile,
+                             Size buffer_size,
+                             cuco::pair<ProbeKey, Key>* buffer,
+                             AtomicT* counter,
+                             OutputIt1 output_probe,
+                             OutputIt2 output_match)
+{
+  Size offset;
+  auto i = tile.thread_rank();
+
+#if defined(CUCO_HAS_CG_INVOKE_ONE)
+  offset = cooperative_groups::invoke_one_broadcast(
+    tile, [&]() { return counter->fetch_add(buffer_size, cuda::std::memory_order_relaxed); });
+#else
+  if (i == 0) { offset = counter->fetch_add(buffer_size, cuda::std::memory_order_relaxed); }
+  offset = tile.shfl(offset, 0);
+#endif
```

Comment:
Is there a performance difference between the two? How is `invoke_one_broadcast` implemented?

Reply:
> Is there a performance difference between the two?

No visible runtime difference but `invoke_one` indeed requires more instructions https://godbolt.org/z/8Y3eeT11j

Reply:
Ha, if we `invoke_one` using a `thread_block_tile` instead of a whole CTA we actually emit less instructions. Notice the `ELECT` instruction: https://godbolt.org/z/q36dYEoTc

Reply:
https://godbolt.org/z/G33zE5aGx

Reply:
It saves more with `invoke_one_broadcast` since shared memory is no longer needed

Reply:
OK, so for the new `invoke_one`/`invoke_one_broadcast` in general:

- tile works better than block
- newer arch like `sm_90` works better than old `sm_70`